md5/md5.go

```diff
@@ -23,10 +25,18 @@ func NewFileContentsSummer(filepath string) Summer {
 }
 
 func (f fileContentsSummer) Sum() (string, error) {
-	fileContents, err := ioutil.ReadFile(f.filepath)
+	fileToSum, err := os.Open(f.filepath)
 	if err != nil {
 		return "", err
 	}
+	defer fileToSum.Close()
 
-	return fmt.Sprintf("%x", md5.Sum(fileContents)), nil
+	fileReader := bufio.NewReader(fileToSum)
+	hash := md5.New()
+	_, err = io.Copy(hash, fileReader)
```

Comment:
Hey @mmb you can actually pass the `fileToSum` directly to `io.Copy` if you so chose. The only reason to have that buffer there is if you are going to make a buffer of a particular size to start (may possibly speed up the `io.Copy` step if the buffer is largeish).